config/src/config/mod.rs

```diff
@@ -145,6 +146,21 @@ impl fmt::Display for RoleType {
     }
 }
 
+#[derive(Debug)]
+pub struct ParseRoleError(String);
+
+impl fmt::Display for ParseRoleError {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(fmt, "{}: {}", self.description(), &self.0)
+    }
+}
+
+impl std::error::Error for ParseRoleError {
+    fn description(&self) -> &str {
+        "Invalid node role"
+    }
```

Comment:
The description method is deprecated and generally should not be implemented in new code.

Reply:
I have a set of commits which are converting everything to use anyhow and thiserror, i'll get rid of this there.